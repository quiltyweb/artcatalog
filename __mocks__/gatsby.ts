import React from "react";

const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(({ to, ...rest }) =>
    React.createElement("a", {
      ...rest,
      href: to,
    })
  ),
  Slice: jest.fn().mockImplementation(({ alias, ...rest }) =>
    React.createElement("div", {
      ...rest,
      "data-test-slice-alias": alias,
    })
  ),
  StaticQuery: jest.fn(),
};
