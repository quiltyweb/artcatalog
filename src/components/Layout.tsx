import * as React from "react";
import Footer from "./Footer";
import { Grid, GridItem } from "@chakra-ui/react";
import Nav from "./Nav";
import styled from "styled-components";
import { useLayoutConsumer } from "../context/LayoutContext";

type LayoutProps = {
  children: React.ReactNode;
};

const VisuallyHiddenH1 = styled.h1`
  border: 0;
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  white-space: nowrap;
`;

import "swiper/css";
import "swiper/css/navigation";
import "../styles/homePageSlider.css";

const Layout: React.FunctionComponent<LayoutProps> = ({
  children,
}): React.ReactElement => {
  const layoutData = useLayoutConsumer();
  return (
    <Grid
      gridTemplateRows={"auto"}
      gridTemplateColumns={"1fr 1fr 1fr 1fr "}
      templateAreas={`"banner banner banner banner"
                      "header header header header"
                      "main main main main"
                      "footer footer footer footer"`}
      gridAutoFlow="row"
      backgroundColor="white"
      color="black"
      fontWeight="normal"
      justifyItems="center"
    >
      <GridItem
        as="header"
        gap="1rem"
        color="#FFFFFF"
        bg={"blackAlpha.900"}
        area={"header"}
        px={7}
        py={3}
        width="100%"
      >
        <Nav
          site={layoutData?.site}
          allShopifyCollection={layoutData?.allShopifyCollection}
        />
      </GridItem>

      <GridItem
        id="main"
        as="main"
        color="#050505"
        bg="white"
        area={"main"}
        w="100%"
        margin="0 auto"
      >
        <VisuallyHiddenH1>Brushella, Art and Home Decor</VisuallyHiddenH1>
        {children}
      </GridItem>
      <GridItem
        as="footer"
        area={"footer"}
        justifySelf="center"
        width={"100%"}
        maxWidth="1200px"
        marginTop="6"
      >
        <Footer legalContent={layoutData?.adminshopify.legalContent} />
      </GridItem>
    </Grid>
  );
};

export default Layout;
