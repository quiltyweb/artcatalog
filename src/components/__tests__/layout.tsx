import React from 'react';
import renderer from 'react-test-renderer';
import * as Gatsby from 'gatsby';
import Layout from '../layout';

beforeEach(() => {
  jest.clearAllMocks();
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('Layout', () => {
  it('renders correctly', () => {
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: 'My Title',
        },
      },
    }));

    const tree = renderer
      .create(
        <Layout pageTitle="hello">
          <p>some content</p>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
