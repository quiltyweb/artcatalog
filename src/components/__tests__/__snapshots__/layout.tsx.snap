// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<main
  className="sc-bdnxRM eevvKH"
>
  <title>
    hello
     | 
    My Title
     
  </title>
  <p>
    My Title
  </p>
  <nav>
    <ul
      className="nav-links"
    >
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link-text"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link-text"
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
  </nav>
  <h1
    className="heading"
  >
    hello
  </h1>
  <p>
    some content
  </p>
</main>
`;
